### PR TITLE
TC_14.002.05 Header Seach box Search functionality Case-sensitive fun…

### PR DESCRIPTION
**Pull request by Pavel-Mi**
https://trello.com/c/5x10hhJm/431-tc1400205-header-seach-box-search-functionality-case-sensitive-functionality-check-casesensitiveoff-lowercase

https://trello.com/c/d2RjZDFl/432-tc1400206-header-search-box-search-functionality-case-sensitive-functionality-check-casesensitiveoff-uppercase

### CODE REVIEW

File: src/test/java/school/redrover/SearchBox1Test.java [hunk 1]
```diff
@@ -67,6 +65,28 @@ public void testCaseSensitiveOnLowercaseInput() {
         Assert.assertEquals(getDriver().findElement(By.cssSelector("[class$='__content']")).getText(),
                 "Log Recorders");
     }
+
+    @Test
+    public void testCaseSensitiveOffLowercaseInput() {
+        turnOnInsensitiveSearch();
+
+        getDriver().findElement(By.id("search-box")).sendKeys("log");
```
Comment (Lokankara) [inline src/test/java/school/redrover/SearchBox1Test.java:73]: DRY

File: src/test/java/school/redrover/SearchBox1Test.java [hunk 2]
```diff
@@ -67,6 +65,28 @@ public void testCaseSensitiveOnLowercaseInput() {
...
+
+    @Test
+    public void testCaseSensitiveOffUppercaseInput() {
+        turnOnInsensitiveSearch();
+
+        getDriver().findElement(By.id("search-box")).sendKeys("loG");
+        getDriver().findElement(By.id("search-box")).sendKeys(Keys.ENTER);
+
+        Assert.assertEquals(getDriver().findElement(By.cssSelector("[class$='__content']")).getText(),
+                "Log Recorders");
```
Comment (Lokankara) [inline src/test/java/school/redrover/SearchBox1Test.java:88]: the expected result is hard-coded and repeated 3 times
you can use static var

File: src/test/java/school/redrover/SearchBox1Test.java [hunk 3]
```diff
@@ -9,40 +9,38 @@
...
+        return new Actions(getDriver());
     }
 
-    private void turnOffInsensitiveSearch() {
+    private void goToConfigure() {
         getDriver().findElement(By.xpath("//div[3]/a[1]")).click();
         getDriver().findElement(By.cssSelector("[href$='configure']")).click();
+    }
+
+    private void checkboxClick() {
```
Comment (Lokankara) [inline src/test/java/school/redrover/SearchBox1Test.java:26]: Your method not only clicks on the item, but also scrolls, sends, and moves to

File: src/test/java/school/redrover/SearchBox1Test.java [hunk 4]
```diff
@@ -67,6 +65,28 @@ public void testCaseSensitiveOnLowercaseInput() {
...
+        getDriver().findElement(By.id("search-box")).sendKeys("log");
+        getDriver().findElement(By.id("search-box")).sendKeys(Keys.ENTER);
+
+        Assert.assertEquals(getDriver().findElement(By.cssSelector("[class$='__content']")).getText(),
+                "Log Recorders");
+    }
+
+    @Test
+    public void testCaseSensitiveOffUppercaseInput() {
+        turnOnInsensitiveSearch();
```
Comment (Lokankara) [inline src/test/java/school/redrover/SearchBox1Test.java:82]: ` turnOnInsensitiveSearch()` and  `turnOffInsensitiveSearch()` have similar code
Perhaps it is better to implement `turnInsensitiveSearch(boolean flag)`

File: src/test/java/school/redrover/SearchBox1Test.java [hunk 5]
```diff
@@ -9,40 +9,38 @@
...
     }
 
-    private void turnOffInsensitiveSearch() {
+    private void goToConfigure() {
         getDriver().findElement(By.xpath("//div[3]/a[1]")).click();
         getDriver().findElement(By.cssSelector("[href$='configure']")).click();
+    }
+
+    private void checkboxClick() {
+        getActions().scrollToElement(getDriver().findElement(By.xpath("//div[3]/div[2]/span")))
```
Comment (Lokankara) [inline src/test/java/school/redrover/SearchBox1Test.java:27]: Using absolute XPaths like `//div[3]/div[2]/span` is not considered a best practice

**Review by Lokankara — APPROVED** [review]

**Review by HelenMyronyuk — APPROVED** [review]

**Pull request merged**